internal/pkg/core/plugin/service/status_watcher.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package service
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/kardianos/service"
+)
+
+const (
+	defaultCheckInterval = 10 * time.Second
+	defaultCheckDuration = 3 * time.Minute
+)
+
+type statusCheckResult struct {
+	Status service.Status
+	Err    error
+}
+
+type serviceWatcher struct {
```

Comment:
None of this seems to be used anymore, is it worth keeping the code or should we delete it?